### DISCUSSION TITLE
Make it possible to setup secure schemes on a per route basis (different secure schemes)

### DISCUSSION THREAD

**Discussion by nuxwin** [dedoc/scramble#293] [Q&A]
Good morning,

Right now, and as per documentation, it is possible to setup default secure scheme for our API, as excluding some methods using the `@unauthenticated` annotation (tag).

In our API, we've plenty of routes, some of them protected using sanctum, and other using stateless HTTP basic authentication.

Could it be possible to add support so one can specify which routes are protected by one or another secure scheme ?

Thank you.

**Comment by nuxwin** [comment]
@romalytvynenko There is a problem with my question  ? I permit to ask you due to your emoji reaction ... You seem suprised of suh a request ...

The fact is that actually, we cannot say : 

This route (or group of routes) is protected by sanctum, while this route (or group of routes) is protected by HTTP basic auth ...

We should be able to say :

1. Default security scheme is that or that.
2. This route or group of routes is protected by that or that security scheme which is not the default.
3. This route or group of routes isn't protected by any secury scheme.

For our API we have even a more complicated security scheme, where a route give restricted information when the requesting user is not authenticated and not an administrator (role), while more sensible information are given when the user is authenticated and is an administrator.

Does I'm clear enough ?

**Reply by romalytvynenko** [reply]
Hey @nuxwin 

Thanks for more context it definitely helps!

P.S.: Not surprised at all, I put eyes emoji when I think the issue is interesting or relevant and I will definitely look into it.
